File: python/semantic_kernel/exceptions/content_exceptions.py
# Copyright (c) Microsoft. All rights reserved.

from semantic_kernel.exceptions.kernel_exceptions import KernelException


class ContentException(KernelException):
    """Base class for all content exceptions."""


class ContentInitializationError(ContentException):
    """An error occurred while initializing the content."""


class ContentSerializationError(ContentException):
    """An error occurred while serializing the content."""


class ContentAdditionException(ContentException):
    """An error occurred while adding content."""


class FunctionCallInvalidNameException(ContentException):
    """An error occurred while validating the function name."""


class FunctionCallInvalidArgumentsException(ContentException):
    """An error occurred while validating the function arguments."""


__all__ = [
    "ContentAdditionException",
    "ContentException",
    "ContentInitializationError",
    "ContentSerializationError",
    "FunctionCallInvalidArgumentsException",
    "FunctionCallInvalidNameException",
]

File: nodejs/semantic-kernel/exceptions/content-exceptions.ts
/**
 * Base class for all content exceptions.
 */
export class ContentException extends Error {
  constructor(message: string) {
    super(message)
    this.name = 'ContentException'
  }
}

/**
 * An error occurred while initializing the content.
 */
export class ContentInitializationError extends ContentException {
  constructor(message: string) {
    super(message)
    this.name = 'ContentInitializationError'
  }
}

/**
 * An error occurred while serializing the content.
 */
export class ContentSerializationError extends ContentException {
  constructor(message: string) {
    super(message)
    this.name = 'ContentSerializationError'
  }
}

/**
 * An error occurred while adding content.
 */
export class ContentAdditionException extends ContentException {
  constructor(message: string) {
    super(message)
    this.name = 'ContentAdditionException'
  }
}

/**
 * An error occurred while validating the function name.
 */
export class FunctionCallInvalidNameException extends ContentException {
  constructor(message: string) {
    super(message)
    this.name = 'FunctionCallInvalidNameException'
  }
}

/**
 * An error occurred while validating the function arguments.
 */
export class FunctionCallInvalidArgumentsException extends ContentException {
  constructor(message: string) {
    super(message)
    this.name = 'FunctionCallInvalidArgumentsException'
  }
}

/**
 * An error occurred while reducing chat history.
 */
export class ChatHistoryReducerException extends ContentException {
  constructor(message: string) {
    super(message)
    this.name = 'ChatHistoryReducerException'
  }
}
